test/clients/openshift/client.go

```diff
@@ -67,80 +72,209 @@ func newClientFromKubeConfig(kc *api.Config) (*Client, error) {
 	return newClientFromRestConfig(restconfig), nil
 }
 
-func clientFromEnv() (*Client, error) {
+func clientFromEnv() (*client, error) {
 	restconfig, err := clientcmd.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
 	if err != nil {
 		return nil, err
 	}
 	return newClientFromRestConfig(restconfig), nil
 }
 
-func NewAzureClusterReaderClient() (*Client, error) {
+func newAzureClusterReaderClient(cs *internalapi.OpenShiftManagedCluster) (*client, error) {
 	if os.Getenv("KUBECONFIG") != "" {
 		return clientFromEnv()
 	}
-	dataDir, err := fakerp.FindDirectory(fakerp.DataDirectory)
-	if err != nil {
-		return nil, err
-	}
-	cs, err := managedcluster.ReadConfig(filepath.Join(dataDir, "containerservice.yaml"))
-	if err != nil {
-		return nil, err
-	}
 
 	var kc api.Config
-	err = latest.Scheme.Convert(cs.Config.AzureClusterReaderKubeconfig, &kc, nil)
+	err := latest.Scheme.Convert(cs.Config.AzureClusterReaderKubeconfig, &kc, nil)
 	if err != nil {
 		return nil, err
 	}
 
 	return newClientFromKubeConfig(&kc)
 }
 
-func NewCustomerReaderClient() (*Client, error) {
+func newCustomerReaderClient(cs *internalapi.OpenShiftManagedCluster) (*client, error) {
 	if os.Getenv("KUBECONFIG") != "" {
```

Comment:
we probably don't need these anymore as you'll be able to pass the cluster struct to the client to get all internals setup correctly. Can you confirm @sozercan